tests/e2e/080_vrf_test.go

```diff
@@ -455,3 +461,169 @@ func TestVRFRoutes(t *testing.T) {
+	Expect(ctx.GetValueMetadata(iface2, kvs.CachedView)).To(
+		HaveKeyWithValue(BeEquivalentTo("VrfMasterIf"), BeEquivalentTo(vrfName)))
+
+	// create vrfIface1 but do not put it into VRF yet
+	err = ifHandler.AddDummyInterface(vrfIface1HostName)
+	Expect(err).To(BeNil())
+	err = ifHandler.SetInterfaceUp(vrfIface1HostName)
+	Expect(err).To(BeNil())
+
+	Eventually(ctx.GetValueStateClb(existingIface1)).Should(Equal(kvscheduler.ValueState_CONFIGURED))
+	Expect(ctx.GetValueMetadata(existingIface1, kvs.CachedView)).To(
+		HaveKeyWithValue(BeEquivalentTo("VrfMasterIf"), BeEquivalentTo(vrfName)))
+	Expect(ctx.GetDerivedValueState(existingIface1, iface1InVrfKey)).To(Equal(kvscheduler.ValueState_PENDING))
+
+	// put interface into VRF (without IPs for now)
+	err = ifHandler.PutInterfaceIntoVRF(vrfIface1HostName, vrfHostName)
+	Expect(err).To(BeNil())
+
+	Eventually(ctx.GetDerivedValueStateClb(existingIface1, iface1InVrfKey)).
+		Should(Equal(kvscheduler.ValueState_CONFIGURED))
+	Consistently(ctx.GetDerivedValueStateClb(existingIface1, ipAddr1Key)).
+		Should(Equal(kvscheduler.ValueState_PENDING))
+	Consistently(ctx.GetDerivedValueStateClb(existingIface1, ipAddr2Key)).
+		Should(Equal(kvscheduler.ValueState_PENDING))
+
+	// re-check metadata after resync
+	Expect(ctx.AgentInSync()).To(BeTrue())
+	Expect(ctx.GetValueMetadata(existingVrf, kvs.CachedView)).To(
+		HaveKeyWithValue(BeEquivalentTo("VrfDevRT"), BeEquivalentTo(vrfRT)))
+	Expect(ctx.GetValueMetadata(existingIface1, kvs.CachedView)).To(
+		HaveKeyWithValue(BeEquivalentTo("VrfMasterIf"), BeEquivalentTo(vrfName)))
+	Expect(ctx.GetValueMetadata(iface2, kvs.CachedView)).To(
+		HaveKeyWithValue(BeEquivalentTo("VrfMasterIf"), BeEquivalentTo(vrfName)))
+
+	// add ipAddr1
+	ipAddr, _, err := utils.ParseIPAddr(ipAddr1+netMask, nil)
+	Expect(err).ToNot(HaveOccurred())
+	err = ifHandler.AddInterfaceIP(vrfIface1HostName, ipAddr)
+	Expect(err).ToNot(HaveOccurred())
+
+	Eventually(ctx.GetDerivedValueStateClb(existingIface1, ipAddr1Key)).
+		Should(Equal(kvscheduler.ValueState_CONFIGURED))
+	Consistently(ctx.GetDerivedValueStateClb(existingIface1, ipAddr2Key)).
+		Should(Equal(kvscheduler.ValueState_PENDING))
+	Expect(ctx.AgentInSync()).To(BeTrue())
+
+	// add ipAddr2
+	ipAddr, _, err = utils.ParseIPAddr(ipAddr2+netMask, nil)
+	Expect(err).ToNot(HaveOccurred())
+	err = ifHandler.AddInterfaceIP(vrfIface1HostName, ipAddr)
+	Expect(err).ToNot(HaveOccurred())
+
+	Eventually(ctx.GetDerivedValueStateClb(existingIface1, ipAddr1Key)).
+		Should(Equal(kvscheduler.ValueState_CONFIGURED))
+	Eventually(ctx.GetDerivedValueStateClb(existingIface1, ipAddr2Key)).
+		Should(Equal(kvscheduler.ValueState_CONFIGURED))
+	Expect(ctx.AgentInSync()).To(BeTrue())
+
+	// cleanup
+	req := ctx.GenericClient().ChangeRequest()
+	err = req.Delete(
+		existingVrf,
+		existingIface1,
+		iface2,
+	).Send(context.Background())
+	Expect(err).ToNot(HaveOccurred())
+	err = ifHandler.DeleteInterface(vrfIface1HostName)
```

Comment:
Why even store the error in a variable if it won't be checked? Better to either put `_ = ...` or check the error.

Reply:
good catch, will add the forgotten check

Reply:
fixed